modules/imgproc/src/intersection.cpp

```diff
@@ -174,17 +165,20 @@ static int _rotatedRectangleIntersection( const RotatedRect& rect1, const Rotate
 
         for( int j = 0; j < 4; j++ )
         {
-            float normalizationScale  = vec2[j].x*vec2[j].x+vec2[j].y*vec2[j].y;
-            normalizationScale  = (normalizationScale < 1e-12f) ? 1.f : 1.f/normalizationScale;
             // line equation: Ax + By + C = 0
             // see which side of the line this point is at
-            const float A = -vec2[j].y*normalizationScale ;
-            const float B = vec2[j].x*normalizationScale ;
-            const float C = -(A*pts2[j].x + B*pts2[j].y);
-
-            const float s = A*x + B*y + C;
-
-            if( s >= 0 )
+            ////base algorithm
```

Comment:
Code will be hard to understand without the s=A*x+B*y+C explanations. Shouldn't we keep track of the underlying meaning ?

Reply:
Maybe something like this then:
```.cpp
// line equation: Ax + By + C = 0 where
// A = -vec2[j].y ; B = vec2[j].x ; C = -(A * pts2[j].x + B * pts2[j].y)
// check which side of the line this point is at
// A*x + B*y + C <> 0
// + computation reordered for better numerical stability
const bool isSPositive = (vec2[j].y*(pts2[j].x-x) >= vec2[j].x*(pts2[j].y-y));
```
And since the same code and comment is repeated in two places it can be extracted into an inline function (e.g. `isOnPositiveSide(const Point2f & line_vec, const Point2f & line_pt, const Point2f & pt)`).